### PR TITLE
Adds test assertions for filtering distributions by repository.

### PR DESCRIPTION
**Pull request by dkliban**
fixes: #881

Required PR: https://github.com/pulp/pulpcore/pull/3503

### CODE REVIEW

File: pulp_file/tests/functional/api/test_auto_publish.py [hunk 1]
```diff
@@ -84,3 +84,10 @@ def test_auto_publish_and_distribution(
     assert file_pub_api_client.list(repository_version=repo.latest_version_href).count == 1
     assert len(files_added) == 1
     assert list(files_added)[0][1] == file_random_content_unit.sha256
+
+    # Assert that filtering distributions by repository is possible
+    distros = file_distro_api_client.list(repository=repo.pulp_href).results
```
Comment (ipanova) [inline pulp_file/tests/functional/api/test_auto_publish.py:89]: can you add a negative test where you filter by a non existent pulp-repo-href to ensure that  the result list will be empty?

Comment (dkliban) [reply pulp_file/tests/functional/api/test_auto_publish.py:89]: I added such an assertion.

**Review by mdellweg — APPROVED** [review]

**Review by ipanova — APPROVED** [review]
